.github/workflows/build-test-distribute.yaml

```diff
@@ -160,6 +246,109 @@ jobs:
           trap on_exit EXIT
           make docker/push
           make docker/manifest
+          manifests=$(make manifests/release/show | jq -R -c -S 'split("\n")') # Create JSON array for release manifests needed for signing
+          echo "$published_manifests=${manifests}"
+          echo "manifests=${manifests}" >> "$GITHUB_OUTPUT"
+  # Keyless container image signing 
+  # This step will sign docker manifest image recursively for all platforms by digest after publishing the image
+  # Publishes all signatures to the "kumahq/notary" public registry until TODO above
+  # Record permanent entry to Sigstore Public Rekor - Transparency log
+  # Note:
+  #  For private repos, no good solution is available for transparency requirement for SLSA.
+  #  Hence exposing repo name and workflow details and commit details by leveraging public sigstore instance
+  #  Generate provenance and image signing separately for every image after publishing. 
+  # TODO:
+  #   Split sigining step and notary registries to avoid customers viewing signatures of internal images
+  #     Official image signatures: kumahq/notary
+  #     Internal/Preview image signatures:kumahq/notary-internal
+  sign-image-manifests:
+    permissions:
+      id-token: write # needed for signing the images with GitHub OIDC Token
+    needs: ["metadata", "distribute-images"]
+    if: >
+      github.repository_owner == 'Kong'  && needs.metadata.outputs.ALLOW_PUSH == 'true'
+
+    runs-on: ubuntu-latest-kong
+    strategy:
+      fail-fast: false
+      matrix:
+        image_manifest: "${{ fromJSON(needs.distribute-images.outputs.published_image_manifests) }}"
+    steps:
+      - name: Install regctl
+        uses: regclient/actions/regctl-installer@main
+      - name: Fetch manifest digest
+        id: manifest_meta
+        run: |
+          echo "Fetching image manifest digest for ${matrix.image_manifest}"
+          echo "digest=$(regctl image digest ${{ matrix.image_manifest }})" >> $GITHUB_OUTPUT
+          echo "label=$(echo {{matrix.image_manifest}} | rev | cut -d "/" -f 1 | rev | cut -d ":" -f1)" >> "$GITHUB_OUTPUT"
+          echo "${tar_name}"
+      - run: |
+          make docker/login
+      - name: sign image manifest
+        id: attest
+        uses: Kong/public-shared-actions/security-actions/sign-docker-image@4119b418bba4462593b5238718772356f84f67ee
+        with:
+          image_digest: ${{ steps.manifest_meta.outputs.digest }}
+          tags: ${{ matrix.image_manifest }}
+          signature_registry: ${{ env.NOTARY_REPOSITORY }}
+        env:
+          NOTARY_REPOSITORY: kumahq/notary # TODO
```

Comment:
Ideally extract this at a high level to make it easier to override in KM

Reply:
done. Point this to "{{github.repository_owner}}/notary" since docker hub org is the same as GitHub org to avoid overriding for KM